iree/tools/vulkan_gui_util.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// IREE's Vulkan HAL is built with VK_NO_PROTOTYPES so Vulkan can be loaded
+// dynamically. Using utilities defined in this header means to link against
+// the Vulkan SDK statically, so we want prototypes to be included.
+#undef VK_NO_PROTOTYPES
+
+#include <SDL.h>
+#include <SDL_vulkan.h>
+#include <vulkan/vulkan.h>
+
+#include <vector>
+
+#include "iree/hal/vulkan/api.h"
+
+// NOTE: order matters here, imgui must come first:
+#include "third_party/dear_imgui/imgui.h"
+// NOTE: must follow imgui.h:
+#include "third_party/dear_imgui/examples/imgui_impl_sdl.h"
+#include "third_party/dear_imgui/examples/imgui_impl_vulkan.h"
+
+namespace iree {
+
+// Returns the names of the Vulkan instance layers needed for the given IREE
+// |vulkan_features|.
+std::vector<const char*> GetInstanceLayers(
+    iree_hal_vulkan_features_t vulkan_features);
+
+// Returns the names of the Vulkan instance extensions needed for the given IREE
+// |vulkan_features|.
+std::vector<const char*> GetInstanceExtensions(
+    SDL_Window* window, iree_hal_vulkan_features_t vulkan_features);
+
+// Initializes the Vulkan environment with the given |vulkan_features| and
+// layers/extensions, and writes various Vulkan handles.
```

Comment:
How are failures handled? Assert fail, or return nullptrs? (Assert is fine for sample code, good to document it here though)